pkg/services/cloudmigration/resource_dependency.go

```diff
@@ -0,0 +1,74 @@
+package cloudmigration
+
+import (
+	"github.com/grafana/grafana/pkg/apimachinery/errutil"
+)
+
+var (
+	ErrDuplicateResourceType = errutil.BadRequest("cloudmigrations.duplicateResourceType")
+	ErrUnknownResourceType   = errutil.BadRequest("cloudmigrations.unknownResourceType")
+	ErrMissingDependency     = errutil.BadRequest("cloudmigrations.missingDependency")
+)
+
+// DependencyMap is a map of resource types to their direct dependencies.
+type DependencyMap map[MigrateDataType][]MigrateDataType
+
+// ResourceDependency is a map of resource types to their direct dependencies.
+// This is used to determine which resources can be filtered out from the snapshot without breaking the dependency chain.
+var ResourceDependency = DependencyMap{
+	PluginDataType:           nil,
+	FolderDataType:           nil,
+	DatasourceDataType:       {PluginDataType},
+	LibraryElementDataType:   {FolderDataType},
+	DashboardDataType:        {FolderDataType, DatasourceDataType, LibraryElementDataType},
+	MuteTimingType:           nil,
+	NotificationTemplateType: nil,
+	ContactPointType:         {NotificationTemplateType},
+	NotificationPolicyType:   {ContactPointType},
+	AlertRuleType:            {DatasourceDataType, FolderDataType, DashboardDataType, MuteTimingType, ContactPointType, NotificationPolicyType},
+	AlertRuleGroupType:       {AlertRuleType},
+}
+
+// Parse a raw slice of resource types and returns a set of them if it has all correct dependencies.
+func (depMap DependencyMap) Parse(rawInput []MigrateDataType) (ResourceTypes, error) {
+	// Clean up any possible duplicates.
+	input := make(ResourceTypes, len(rawInput))
+	for _, resourceType := range rawInput {
+		if _, exists := input[resourceType]; exists {
+			return nil, ErrDuplicateResourceType.Errorf("duplicate resource type found: %v", resourceType)
```

Comment:
Could we add a todo or reminder somewhere to make sure our swagger documentation adequately describes these error conditions?

Reply:
Oh I am looking at the second PR now, nvm, will comment there if necessary!

Reply:
Sure, I will not merge the first PR to main, will make it kind of like a feature branch I think..